Src/IronPython/Runtime/Operations/FloatOps.cs

```diff
@@ -1017,6 +1017,29 @@ public static partial class DoubleOps {
                     throw PythonOps.ValueError("__setformat__() argument 1 must be 'double' or 'float'");
             }
         }
+
+        public static double __round__(double self) {
+            // as of Python 3 rounding is to the nearest even number, not away from zero
+            return MathUtils.Round(self, 0, MidpointRounding.ToEven);
+        }
+
+        public static double __round__(double self, int ndigits) {
+            return PythonOps.CheckMath(self, MathUtils.Round(self, ndigits, MidpointRounding.ToEven));
```

Comment:
Interesting, is there a test covering the case where it overflows to infinity?

Reply:
There is now. Obviously the previous behaviour was all wrong. Good catch!